fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/CacheDictManager.java

```diff
@@ -260,6 +260,12 @@ public void updateGlobalDict(long tableId, String columnName, long versionTime)
                 if (columnOptional.isPresent()) {
                     ColumnDict columnDict = columnOptional.get();
                     long lastVersion = columnDict.getVersionTime();
+                    long dictCollectVersion = columnDict.getCollectedVersionTime();
+                    if (collectVersion != dictCollectVersion) {
```

Comment:
is there exists a race condtion like this?
the query choose the latest version of dict, just after that, the this version is overtaken by insert-into transaction which add more new items to the dict-encoded columns.

Reply:
updateGlobalDict will be protected by DB.Lock. it will will called in transaction replay thread.

Reply:
However, there may be a situation where our planner is not protected by a lock, so the dictionaries obtained may still be inconsistent. One possible solution is to eliminate all truncates from this table when doing truncate operations